Dockerfile

```diff
@@ -0,0 +1,29 @@
+FROM python:3.8-slim-buster AS intermediate
+ARG PYPI_INDEX_URL
+ENV URL=${PYPI_INDEX_URL}
+# comes as an argument for 'docker build --build-arg ... ' 
+RUN python -m pip install --upgrade pip
+RUN mkdir /root/.pip && touch /root/.pip/pip.conf && \
+    printf "[global]\nindex-url = ${URL}\n" > /root/.pip/pip.conf && \
+    cat /root/.pip/pip.conf
+RUN pip install myWallet==0.2
+
+FROM python:3.8-slim-buster
+RUN useradd --create-home appuser
+RUN chown -R appuser /home/appuser
+# path where pip looks for packages is '/usr/local/lib/python3.8/site-packages'
+COPY --from=intermediate /usr/local/lib/python3.8/site-packages /usr/local/lib/python3.8/site-packages 
+WORKDIR /home/appuser
+USER appuser
+COPY --chown=appuser:appuser requirements.txt requirements.txt
+RUN python -m pip install --upgrade pip && \
+    pip install myWallet==0.2  && \
```

Comment:
is it neccessary to install myWallet==0.2 as root? you have the dependancy in requirements.txt

Reply:
As appuser. not root.

Reply:
Interesting question as logs shows that `Requirement already satisfied: myWallet==0.2 in /usr/local/lib/python3.8/site-packages` , I was just curious if the extra step is necessary - I might be wrong but will investigate and will let you know

Reply:
It is obvious if we installed that package before others [available globally in world PyPi]. see the trick with a 'pip.conf' file substitution.